std/file.d

```diff
@@ -3196,9 +3207,9 @@ unittest
     import std.algorithm, std.file; // issue 14817
     auto t1 = deleteme, t2 = deleteme~"2";
     scope(exit) foreach (t; [t1, t2]) if (t.exists) t.remove();
-    write(t1, "1");
+    write(t1, "11");
```

Comment:
To test that the file is actually truncated, i.e. that we don't just overwrite the first `fileSize(source)` bytes.